python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py

```diff
@@ -1,40 +1,58 @@
 # Copyright (c) Microsoft. All rights reserved.
 
-import json
 import logging
 from collections.abc import AsyncGenerator
 from typing import Any
 
-import aiohttp
-from pydantic import HttpUrl
+from ollama import AsyncClient
+from pydantic import ValidationError
 
 from semantic_kernel.connectors.ai.chat_completion_client_base import ChatCompletionClientBase
 from semantic_kernel.connectors.ai.ollama.ollama_prompt_execution_settings import OllamaChatPromptExecutionSettings
-from semantic_kernel.connectors.ai.ollama.utils import AsyncSession
+from semantic_kernel.connectors.ai.ollama.ollama_settings import OllamaSettings
+from semantic_kernel.connectors.ai.ollama.services.ollama_base import OllamaBase
 from semantic_kernel.connectors.ai.text_completion_client_base import TextCompletionClientBase
 from semantic_kernel.contents import AuthorRole
 from semantic_kernel.contents.chat_history import ChatHistory
 from semantic_kernel.contents.chat_message_content import ChatMessageContent
 from semantic_kernel.contents.streaming_chat_message_content import StreamingChatMessageContent
 from semantic_kernel.contents.streaming_text_content import StreamingTextContent
 from semantic_kernel.contents.text_content import TextContent
+from semantic_kernel.exceptions.service_exceptions import ServiceInitializationError
 
 logger: logging.Logger = logging.getLogger(__name__)
 
 
-class OllamaChatCompletion(TextCompletionClientBase, ChatCompletionClientBase):
+class OllamaChatCompletion(OllamaBase, TextCompletionClientBase, ChatCompletionClientBase):
     """Initializes a new instance of the OllamaChatCompletion class.
 
     Make sure to have the ollama service running either locally or remotely.
-
-    Args:
-        ai_model_id (str): Ollama model name, see https://ollama.ai/library
-        url (Optional[Union[str, HttpUrl]]): URL of the Ollama server, defaults to http://localhost:11434/api/chat
-        session (Optional[aiohttp.ClientSession]): Optional client session to use for requests.
     """
 
-    url: HttpUrl = "http://localhost:11434/api/chat"
-    session: aiohttp.ClientSession | None = None
+    def __init__(
+        self,
+        service_id: str | None = None,
+        ai_model_id: str | None = None,
+        host: str | None = None,
```

Comment:
We're missing the `env_file_path` and `env_file_encoding` args, right?

Reply:
Added.